source/includes/fundamentals/code-snippets/transaction.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/writeconcern"
+)
+
+func main() {
+
+	var uri string
+	if uri = os.Getenv("MONGODB_URI"); uri == "" {
+		log.Fatal("You must set your 'MONGODB_URI' environmental variable. See\n\t https://www.mongodb.com/docs/drivers/go/current/usage-examples/")
+	}
+
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+	if err != nil {
+		panic(err)
+	}
+	defer client.Disconnect(context.TODO())
+
+	database := client.Database("myDB")
+	coll := database.Collection("myColl")
+
+	// start-session
+	wc := writeconcern.New(writeconcern.WMajority())
+	txnOptions := options.Transaction().SetWriteConcern(wc)
+
+	session, err := client.StartSession()
+	if err != nil {
+		panic(err)
+	}
+	defer session.EndSession(context.Background())
+
+	err = mongo.WithSession(context.Background(), session, func(sessCtx mongo.SessionContext) error {
+		if err = session.StartTransaction(txnOptions); err != nil {
```

Comment:
A more common pattern is to use the `Session.WithTransaction` helper that handles creating and committing (or aborting) the transaction. Also, transactions are unnecessary with a single operation (all single-document operations are already atomic operations), so a more realistic scenario is to insert or modify multiple documents in a transaction.

E.g. using `session.WithTransaction` and inserting 3 documents:
```go
result, err = session.WithTransaction(context.TODO(), func(ctx mongo.SessionContext) error {
	result, err := coll.InsertMany(ctx, []interface{}{
		bson.D{{"title", "The Bluest Eye"}, {"author", "Toni Morrison"}},
		bson.D{{"title", "Sula"}, {"author", "Toni Morrison"}},
		bson.D{{"title", "Song of Solomon"}, {"author", "Toni Morrison"}},
	})
	return result, err
}, txnOptions)

fmt.Println(result.InsertedIDs)
```

Is there a reason we want to use the more verbose `StartTransaction` and `CommitTransaction`/`AbortTransaction` API instead of the more terse `WithTransaction`?

Reply:
I thought it would be helpful for users to see the methods in use in the example, but if using `WithTransaction` is a better approach, I can revise the example. For now, I've changed the insert to `InsertMany` to demonstrate a multi-document transaction.

Reply:
Adding an example using `WithTransaction` would be really helpful since that's the way we expect and want most people to use transactions. Keeping this example of how to create/commit/abort transactions manually may also be useful to users who need more control over transactions.